File: src/main/java/com/phase2/javaProject_Phase2/beans/Coupon.java
package com.phase2.javaProject_Phase2.beans;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

import java.sql.Date;
import java.util.List;

@Entity
@Table(name = "coupons")
@Data
@AllArgsConstructor
@NoArgsConstructor

public class Coupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false)
    private Integer company_ID;

    @Column(nullable = false)
    private Integer category_ID;

    @Column(nullable = false) // exception for sql
    @Length(min = 1, max = 400) // exception for advice
    private String title;

    @Column(nullable = false, unique = true, length = 40)
    @Length(min = 10, max = 40)
    private String description;

    @Column(nullable = false)
    private Date start_date;

    @Column(nullable = false)
    private Date end_date;

    @Min(1)
    private Integer amount;

    @Min(1)
    private Double price;

    private String image;



    public Coupon(Integer company_ID, Integer category_ID, String title, String description,
                  Date start_date, Date end_date, Integer amount, Double price, String image) {
        this.company_ID = company_ID;
        this.category_ID = category_ID;
        this.title = title;
        this.description = description;
        this.start_date = start_date;
        this.end_date = end_date;
        this.amount = amount;
        this.price = price;
        this.image = image;
    }
}

File: Frontend/src/Components/Models/Coupon.ts
export class Coupon{
    public id:number;
    public companyId:number;
    public categoryId:string;
    public title: string;
    public description: string;
    public startDate: string;
    public endDate: string;
    public amount: number;
    public price: number;
    public image: string;

    constructor(id:number, companyId:number, categoryId:string, title:string, description:string, 
        startDate:string, endDate:string, amount:number, price:number, image:string){
            this.id = id;
            this.companyId = companyId;
            this.categoryId = categoryId;
            this.title = title;
            this.description = description;
            this.startDate = startDate;
            this.endDate = endDate;
            this.amount = amount;
            this.price = price;
            this.image = image;
        }
}